crates/sui-graphql-rpc/src/context_data/pg_backend.rs

```diff
@@ -91,139 +91,134 @@ impl GenericQueryBuilder<Pg> for PgQueryBuilder {
-                        .filter(tx_senders::dsl::sender.eq(signer.into_vec()))
-                        .select(tx_senders::dsl::tx_sequence_number);
+        // at_checkpoint mutually exclusive with before_ and after_checkpoint
+        if let Some(checkpoint) = filter.at_checkpoint {
+            query =
+                query.filter(transactions::dsl::checkpoint_sequence_number.eq(checkpoint as i64));
+        }
+        if let Some(transaction_ids) = filter.transaction_ids {
+            let digests = transaction_ids
+                .into_iter()
+                .map(|id| Ok::<Vec<u8>, Error>(Digest::from_str(&id)?.into_vec()))
+                .collect::<Result<Vec<_>, _>>()?;
+            query = query.filter(transactions::dsl::transaction_digest.eq_any(digests));
+        }
 
-                    query = query.filter(transactions::dsl::tx_sequence_number.eq_any(subquery));
-                }
-            } else if let Some(sender) = filter.sent_address {
-                let subquery = tx_senders::dsl::tx_senders
-                    .filter(tx_senders::dsl::sender.eq(sender.into_vec()))
-                    .select(tx_senders::dsl::tx_sequence_number);
+        // Queries on foreign tables
+        match (filter.package, filter.module, filter.function) {
+            (Some(p), None, None) => {
+                let subquery = tx_calls::dsl::tx_calls
+                    .filter(tx_calls::dsl::package.eq(p.into_vec()))
+                    .select(tx_calls::dsl::tx_sequence_number);
 
                 query = query.filter(transactions::dsl::tx_sequence_number.eq_any(subquery));
             }
-            if let Some(recipient) = filter.recv_address {
-                let subquery = tx_recipients::dsl::tx_recipients
-                    .filter(tx_recipients::dsl::recipient.eq(recipient.into_vec()))
-                    .select(tx_recipients::dsl::tx_sequence_number);
+            (Some(p), Some(m), None) => {
+                let subquery = tx_calls::dsl::tx_calls
+                    .filter(tx_calls::dsl::package.eq(p.into_vec()))
+                    .filter(tx_calls::dsl::module.eq(m))
+                    .select(tx_calls::dsl::tx_sequence_number);
 
                 query = query.filter(transactions::dsl::tx_sequence_number.eq_any(subquery));
             }
-            if filter.paid_address.is_some() {
-                return Err(Error::Internal(
-                    "Paid address filter not supported".to_string(),
-                ));
+            (Some(p), Some(m), Some(f)) => {
+                let subquery = tx_calls::dsl::tx_calls
+                    .filter(tx_calls::dsl::package.eq(p.into_vec()))
+                    .filter(tx_calls::dsl::module.eq(m))
+                    .filter(tx_calls::dsl::func.eq(f))
+                    .select(tx_calls::dsl::tx_sequence_number);
+
+                query = query.filter(transactions::dsl::tx_sequence_number.eq_any(subquery));
             }
+            _ => {}
```

Comment:
Poor line, it looks lonely.

Reply:
Could make it explicit that it's the `(None, None, None)` case, and leave a comment that it's supposed to be a `nop`?

```suggestion
            (None, None, None) => { /* nop */ }
```